src/Storage/TSStorageManager.m

```diff
@@ -296,7 +307,14 @@ - (NSData *)databasePassword
         [SAMKeychain passwordForService:keychainService account:keychainDBPassAccount error:&keyFetchError];
 
     if (keyFetchError) {
-        // Either this is a new install so there's no existing password to retrieve
```

Comment:
Wait, I take it back, before this second `if`, let's:

Let's:

* Write error.description to the logs.
* Write [UIApplication sharedApplication].applicationState to the logs.
* Flush the logs.

That way its in the logs whether or not we a) raise an exception or b) go to clean install mode.